sklearn/compose/_column_transformer.py

```diff
@@ -628,14 +628,9 @@ def _get_column_indices(X, key):
     """
     n_columns = X.shape[1]
 
-    if _check_key_type(key, int):
-        if isinstance(key, int):
-            return [key]
-        elif isinstance(key, slice):
-            return list(range(n_columns)[key])
-        else:
-            return list(key)
-
+    if (_check_key_type(key, int)
+            or hasattr(key, 'dtype') and np.issubdtype(key.dtype, np.bool_)):
+        return np.atleast_1d(np.arange(n_columns)[key]).tolist()
```

Comment:
Now that I see it, this is a little hard to read.

Maybe add a comment `# Interpret the index like numpy does`

or pull out `idx = np.arange(n_columns)[key]`